doc/ConfigureOptions.md

```diff
@@ -0,0 +1,123 @@
+
+It is common for modules and agents to have various options that they want their
+users to set in the config.json, but users, being inheritly lazy, do not want to
+actually edit a file, so it is nice if they have friendly prompts they can
+answer when adding a module or agent to a local Halibot instance.
+
+This interface enables the object implementer to configure his configuration.
+
+The Configure class
+===================
+
+The `add` command will automatically create a class named `Configurer` that exists
+on the module or agent class and call the `configure` function on that class. As
+long as that class sets up the `config` field of the `Configurer` instance by
+the end of the `configure` function, everything will work hunky dory, but in
+order to make things easier, Halibot provides a class called `HalConfigurer`,
+which you should definitely subclass.
+
+The `HalConfigurer` class provides a method called `option` which prompts the
+user to configure an option. For syntax is as follows:
+
+```python
+HalConfigurer.option(option_type, key=, prompt=, default=)
+```
+
+* `option_type` is an instance of a subclass of `Halibot.Option` that determines
+  how to validate and prompt for the option.
+* `key` is the name of the key in the config to set. If this is not set, the
+  config object is returned from the function and it is the user's
+  responsibility to set the appropriate field in `self.config`. Not setting this
+  is useful for configuring things such as entries of an array within the config.
+* `prompt` is the string to prompt the user with. If this is left blank, the
```

Comment:
I was thinking about that too... but I can see a case where the prompt and the description would be different. Perhaps have a description which provides a default prompt and if you really want to you can set bolth?